assignments/assignment06.rb

```diff
@@ -7,46 +7,110 @@
 
 # implement a PriorityQueue, validate using MiniTest unit tests
 
-class PriorityQueue
+class Priority_Queue
+  attr :list, :nL, :nH, :length
   def initialize
-    # your implementation here
+    @list = []
+    @nL = 0
+    @nH = 0
+    @length = 0
   end
-  def enqueue(item, priority=:medium)
-    # your implementation here
+
+  def enqueue(item, priority = :normal)
+    entry = {item: item, priority: priority}
+    case priority
+    when :high
+      @list.insert(@length - @nH, entry)
+      @nH += 1
+    when :low
+      @list.insert(0, entry)
+      @nL += 1
+    else
+      @list.insert(@nL, entry)
+    end
+    @length += 1
   end
+
   def dequeue
-    # your implementation here
+    if @length > 0
+      last = @list.pop
+      if last[:priority] == :high
```

Comment:
I would refactor the adjustment of the @nH and @nL instances vars into a separate method.